src/backend/distributed/metadata/dependency.c

```diff
@@ -1249,8 +1249,9 @@ IsObjectAddressOwnedByCitus(const ObjectAddress *objectAddress)
 		return false;
 	}
 
-	bool ownedByCitus = extObjectAddress.objectId == citusId;
-	bool ownedByCitusColumnar = extObjectAddress.objectId == citusColumnarId;
+	bool ownedByCitus = OidIsValid(citusId) && extObjectAddress.objectId == citusId;
```

Comment:
We are already checking for this? May be change && to ||?
`if (!OidIsValid(citusId) && !OidIsValid(citusColumnarId))`